From 420bbfe27435857128aaac69b00faf99ac7388e9 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Tue, 30 Jul 2013 12:59:53 +0800
Subject: [PATCH 225/602] ENGR00256988 config: update board config for wifi

Update wifi configs. Now we use virtal device UNITE for both
realtek and atheros wifi.so some svc will need two copy.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 evk_6sl/BoardConfig.mk      | 8 ++++++--
 imx6/etc/init.rc            | 7 +++++++
 imx6/imx6.mk                | 1 +
 sabreauto_6q/BoardConfig.mk | 8 ++++++--
 sabresd_6dq/BoardConfig.mk  | 8 ++++++--
 5 files changed, 26 insertions(+), 6 deletions(-)

diff --git a/evk_6sl/BoardConfig.mk b/evk_6sl/BoardConfig.mk
index 4d85b09..23da170 100644
--- a/evk_6sl/BoardConfig.mk
+++ b/evk_6sl/BoardConfig.mk
@@ -23,10 +23,14 @@ TARGET_KERNEL_MODULES        := \
 BOARD_WPA_SUPPLICANT_DRIVER  := NL80211
 BOARD_HOSTAPD_DRIVER         := NL80211
 
+BOARD_HOSTAPD_PRIVATE_LIB_QCOM              := lib_driver_cmd_qcwcn
+BOARD_WPA_SUPPLICANT_PRIVATE_LIB_QCOM       := lib_driver_cmd_qcwcn
+BOARD_HOSTAPD_PRIVATE_LIB_RTL               := lib_driver_cmd_rtl
+BOARD_WPA_SUPPLICANT_PRIVATE_LIB_RTL        := lib_driver_cmd_rtl
 #for intel vendor
 ifeq ($(BOARD_WLAN_VENDOR),INTEL)
-BOARD_HOSTAPD_PRIVATE_LIB                ?= private_lib_driver_cmd
-BOARD_WPA_SUPPLICANT_PRIVATE_LIB         ?= private_lib_driver_cmd
+BOARD_HOSTAPD_PRIVATE_LIB                := private_lib_driver_cmd
+BOARD_WPA_SUPPLICANT_PRIVATE_LIB         := private_lib_driver_cmd
 WPA_SUPPLICANT_VERSION                   := VER_0_8_X
 HOSTAPD_VERSION                          := VER_0_8_X
 BOARD_WPA_SUPPLICANT_PRIVATE_LIB         := private_lib_driver_cmd_intel
diff --git a/imx6/etc/init.rc b/imx6/etc/init.rc
index 72d1687..5406e54 100755
--- a/imx6/etc/init.rc
+++ b/imx6/etc/init.rc
@@ -660,6 +660,13 @@ service hostapd /system/bin/hostapd /data/misc/wifi/hostapd.conf
     oneshot
     disabled
 
+service rtl_hostapd /system/bin/hostapd /data/misc/wifi/hostapd.conf
+    socket hostapd_wlan0 dgram 660 root wifi
+    user root
+    group wifi
+    oneshot
+    disabled
+
 service ril-daemon /system/bin/rild
     class main
     socket rild stream 660 root radio
diff --git a/imx6/imx6.mk b/imx6/imx6.mk
index 8ac0a92..0cd2c18 100644
--- a/imx6/imx6.mk
+++ b/imx6/imx6.mk
@@ -57,6 +57,7 @@ PRODUCT_PACKAGES += \
 # Wifi AP mode
 PRODUCT_PACKAGES += \
 	hostapd 				\
+	rtl_hostapd 				\
 	hostapd_cli
 
 # keyboard mapping files.
diff --git a/sabreauto_6q/BoardConfig.mk b/sabreauto_6q/BoardConfig.mk
index c6505b5..18a2fa2 100644
--- a/sabreauto_6q/BoardConfig.mk
+++ b/sabreauto_6q/BoardConfig.mk
@@ -38,10 +38,14 @@ TARGET_KERNEL_MODULES        := \
 BOARD_WPA_SUPPLICANT_DRIVER  := NL80211
 BOARD_HOSTAPD_DRIVER         := NL80211
 
+BOARD_HOSTAPD_PRIVATE_LIB_QCOM              := lib_driver_cmd_qcwcn
+BOARD_WPA_SUPPLICANT_PRIVATE_LIB_QCOM       := lib_driver_cmd_qcwcn
+BOARD_HOSTAPD_PRIVATE_LIB_RTL               := lib_driver_cmd_rtl
+BOARD_WPA_SUPPLICANT_PRIVATE_LIB_RTL        := lib_driver_cmd_rtl
 #for intel vendor
 ifeq ($(BOARD_WLAN_VENDOR),INTEL)
-BOARD_HOSTAPD_PRIVATE_LIB                ?= private_lib_driver_cmd
-BOARD_WPA_SUPPLICANT_PRIVATE_LIB         ?= private_lib_driver_cmd
+BOARD_HOSTAPD_PRIVATE_LIB                := private_lib_driver_cmd
+BOARD_WPA_SUPPLICANT_PRIVATE_LIB         := private_lib_driver_cmd
 WPA_SUPPLICANT_VERSION                   := VER_0_8_X
 HOSTAPD_VERSION                          := VER_0_8_X
 BOARD_WPA_SUPPLICANT_PRIVATE_LIB         := private_lib_driver_cmd_intel
diff --git a/sabresd_6dq/BoardConfig.mk b/sabresd_6dq/BoardConfig.mk
index 447b596..a3f315c 100755
--- a/sabresd_6dq/BoardConfig.mk
+++ b/sabresd_6dq/BoardConfig.mk
@@ -33,10 +33,14 @@ TARGET_KERNEL_MODULES        := \
 BOARD_WPA_SUPPLICANT_DRIVER  := NL80211
 BOARD_HOSTAPD_DRIVER         := NL80211
 
+BOARD_HOSTAPD_PRIVATE_LIB_QCOM              := lib_driver_cmd_qcwcn
+BOARD_WPA_SUPPLICANT_PRIVATE_LIB_QCOM       := lib_driver_cmd_qcwcn
+BOARD_HOSTAPD_PRIVATE_LIB_RTL               := lib_driver_cmd_rtl
+BOARD_WPA_SUPPLICANT_PRIVATE_LIB_RTL        := lib_driver_cmd_rtl
 #for intel vendor
 ifeq ($(BOARD_WLAN_VENDOR),INTEL)
-BOARD_HOSTAPD_PRIVATE_LIB                ?= private_lib_driver_cmd
-BOARD_WPA_SUPPLICANT_PRIVATE_LIB         ?= private_lib_driver_cmd
+BOARD_HOSTAPD_PRIVATE_LIB                := private_lib_driver_cmd
+BOARD_WPA_SUPPLICANT_PRIVATE_LIB         := private_lib_driver_cmd
 WPA_SUPPLICANT_VERSION                   := VER_0_8_X
 HOSTAPD_VERSION                          := VER_0_8_X
 BOARD_WPA_SUPPLICANT_PRIVATE_LIB         := private_lib_driver_cmd_intel
-- 
1.8.0

